tests/cstor/snapshot/snapshot_test.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package snapshot
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	pvc "github.com/openebs/maya/pkg/kubernetes/persistentvolumeclaim/v1alpha1"
+	snap "github.com/openebs/maya/pkg/kubernetes/snapshot/v1alpha1"
+	sc "github.com/openebs/maya/pkg/kubernetes/storageclass/v1alpha1"
+	spc "github.com/openebs/maya/pkg/storagepoolclaim/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("[cstor] TEST SNAPSHOT PROVISIONING", func() {
+	var (
+		err      error
+		pvcName  = "cstor-pvc"
+		snapName = "cstorsnapshot"
+	)
+
+	BeforeEach(func() {
+		When("we are creating pool deployment", func() {
+			By("building object of storageclass")
+			scObj, err = sc.NewBuilder().
+				WithName(scName).
+				WithAnnotations(annotations).
+				WithProvisioner(openebsProvisioner).Build()
+			Expect(err).ShouldNot(HaveOccurred(), "while building storageclass obj for storageclass {%s}", scName)
+
+			By("creating storageclass")
+			_, err = ops.SCClient.Create(scObj)
+			Expect(err).To(BeNil(), "while creating storageclass", scName)
+
+			By("building spc object")
+			spcObj = spc.NewBuilder().
+				WithName(spcName).
+				WithDiskType(string(apis.TypeSparseCPV)).
+				WithMaxPool(1).
+				WithOverProvisioning(false).
+				WithPoolType(string(apis.PoolTypeStripedCPV)).
+				Build().Object
+
+			By("creating storagepoolclaim")
+			_, err = ops.SPCClient.Create(spcObj)
+			Expect(err).To(BeNil(), "while creating spc", spcName)
+
+			By("verifying healthy csp count")
+			cspCount := ops.GetHealthyCSPCount(spcName, 1)
+			Expect(cspCount).To(Equal(1), "while checking cstorpool health count")
+
+		})
+	})
+
+	AfterEach(func() {
+		By("deleting resources created for cstor volume snapshot provisioning", func() {
+			By("listing spc")
+			spcList, err = ops.SPCClient.List(metav1.ListOptions{})
```

Comment:
Why do we need to list & delete?
We already have the name of the SPC that needs to be deleted.

Reply:
Yes not required .. removed